rasa_nlu/extractors/entity_synonyms.py

```diff
@@ -74,16 +75,18 @@ def load(cls,
         # type: (...) -> EntitySynonymMapper
 
         meta = model_metadata.for_component(cls.name)
-        file_name = meta.get("synonyms_file", ENTITY_SYNONYMS_FILE_NAME)
-        entity_synonyms_file = os.path.join(model_dir, file_name)
+        file_name = meta.get("synonyms_file")
+        if not file_name:
+            synonyms = None
+            return cls(meta, synonyms)
 
+        entity_synonyms_file = os.path.join(model_dir, file_name)
         if os.path.isfile(entity_synonyms_file):
             synonyms = utils.read_json_file(entity_synonyms_file)
         else:
             synonyms = None
             warnings.warn("Failed to load synonyms file from '{}'"
                           "".format(entity_synonyms_file))
```

Comment:
while we're modifying this part of the code, why is this a warning @tmbo? surely it should fail at that point 😛

Reply:
Although I agree that it should be an error, there are now many models out there that reference this file because of this previous bug - so I'd keep it a warning

Reply:
but won't this be part of 0.14 anyways so it's ok to have a breaking change? :P

Reply:
Let's create a separate issue for this, I'd hate to break model compatibility just for this one - if we need to break it because of other changes anyways than I am onboard.

Reply:
oki doki